pkg/workflows/wasm/v2/runner.go

```diff
@@ -0,0 +1,170 @@
+package wasm
+
+import (
+	"encoding/base64"
+	"fmt"
+	"io"
+	"log/slog"
+	"os"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/workflows/internal/v2/sdkimpl"
+	"google.golang.org/protobuf/proto"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/values"
+	"github.com/smartcontractkit/chainlink-common/pkg/workflows/sdk/v2"
+	sdkpb "github.com/smartcontractkit/chainlink-common/pkg/workflows/sdk/v2/pb"
+	"github.com/smartcontractkit/chainlink-common/pkg/workflows/wasm/v2/pb"
+)
+
+var args = os.Args
+
+func newDonRunner() sdk.DonRunner {
+	drt := &sdkimpl.DonRuntime{RuntimeBase: newRuntime()}
+	return getRunner(&subscriber[sdk.DonRuntime]{}, &runner[sdk.DonRuntime]{runtime: drt, setRuntime: func(id string, config []byte, maxResponseSize uint64) {
+		drt.ExecId = id
+		drt.ConfigBytes = config
+		drt.MaxResponseSize = maxResponseSize
+	}})
+}
+
+func newNodeRunner() sdk.NodeRunner {
+	nrt := &sdkimpl.NodeRuntime{RuntimeBase: newRuntime()}
+	return getRunner(&subscriber[sdk.NodeRuntime]{}, &runner[sdk.NodeRuntime]{runtime: nrt, setRuntime: func(id string, config []byte, maxResponseSize uint64) {
+		nrt.ExecId = id
+		nrt.ConfigBytes = config
+		nrt.MaxResponseSize = maxResponseSize
+	}})
+}
+
+type runner[T any] struct {
+	trigger    *sdkpb.Trigger
+	id         string
+	runtime    T
+	setRuntime func(id string, config []byte, maxResponseSize uint64)
+	config     []byte
+}
+
+var _ sdk.DonRunner = &runner[sdk.DonRuntime]{}
+var _ sdk.NodeRunner = &runner[sdk.NodeRuntime]{}
+
+func (d *runner[T]) Run(args *sdk.WorkflowArgs[T]) {
+	// used to ensure that the export isn't optimized away
+	versionV2()
+	for _, handler := range args.Handlers {
+		// TODO: https://smartcontract-it.atlassian.net/browse/CAPPL-809 multiple of the same trigger registered
+		// The ID field could be changed to tirger-# and we can use the index or similar.
+		if handler.Id() == d.trigger.Id {
+			response, err := handler.Callback()(d.runtime, d.trigger.Payload)
+			execResponse := &pb.ExecutionResult{Id: d.id}
+			if err == nil {
+				wrapped, err := values.Wrap(response)
+				if err != nil {
+					execResponse.Result = &pb.ExecutionResult_Error{Error: err.Error()}
+				} else {
+					execResponse.Result = &pb.ExecutionResult_Value{Value: values.Proto(wrapped)}
+				}
+			} else {
+				execResponse.Result = &pb.ExecutionResult_Error{Error: err.Error()}
+			}
+			marshalled, _ := proto.Marshal(execResponse)
```

Comment:
nit: still handle those errors? (and a few more places below)

Reply:
I don't know what we or the user would do with the error. It shouldn't be possible, given that we made the protos, so we know it'll work with marshal.